mappings/InventoryID-LanguageCode-Glottocode-LanguageName-Source.tsv

```diff
@@ -0,0 +1,3017 @@
+2020	yme	yame1242	Yameo	saphon
+2021	qwa	coro1247	Ancash Quechua	saphon
+2022	qvc	caja1238	Cajamarca Quechua	saphon
+2023	quk	chac1250	Chachapoyas Quechua	saphon
+2024	quz	cusc1236	Cuzco-Collao Quechua	saphon
+2025	quf	lamb1276	Ferreñafe Quechua	saphon
+2026	qvi	imba1240	Imbabura Quichua	saphon
+2027	inb	inga1252	Inga	saphon
+2028	inj	jung1240	Inga	saphon
+2029	qxw	jauj1238	Jauja-Huanca Quechua	saphon
+2030	qvn	nort2980	North Junín Quechua	saphon
+2031	qvn	nort2980	North Junín Quechua	saphon
+2032	quy	ayac1239	Ayacucho Quechua	saphon
+2033	qul	nort2976	Bolivian Quechua	saphon
+2034	qub	hual1241	Huallaga Huánuco Quechua	saphon
+2035	qxn	nort2979	Huaylas-Conchucos Quechua	saphon
+2036	qvs	sanm1289	San Martin Quechua	saphon
+2037	qvn	nort2980	Yaru Quechua	saphon
+2038	qus	sant1432	Santiago del Estero Quechua	saphon
+2039	qvo	napo1242	Napo Quichua	saphon
+2040	qxl	sala1272	Salasca Quechua	saphon
+2041	quw	tena1240	Tena Quechua	saphon
+2042	pid	piar1243	Piaroa	saphon
+2043	slc	sali1298	Sáliba	saphon
+2044	aro	arao1248	Araona	saphon
+2045	cav	cavi1250	Cavineña	saphon
+2046	ese	esee1248	Ese Ejja	saphon
+2047	ese	esee1248	Ese Eja	saphon
+2048	rey	reye1240	Reyesano	saphon
+2049	tna	taca1256	Tacana	saphon
+2050	bsn	bara1380	Barasana-Eduria	saphon
+2051	cub	cube1242	Kubeo	saphon
+2052	des	desa1247	Desano	saphon
+2053	gvc	guan1269	Wanano	saphon
+2054	cbc	cara1272	Karapanã	saphon
+2055	coe	kore1283	Koreguaje	saphon
+2056	myy	macu1260	Macuna	saphon
+2057	ore	orej1242	Máíhɨki	saphon
+2058	pir	pira1254	Piratapuyo	saphon
+2059	sey	seco1241	Secoya	saphon
+2060	snn	sion1247	Siona	saphon
+2061	sri	siri1274	Siriano	saphon
+2062	tnc	tani1257	Tanimuca-Retuarã	saphon
+2063	tav	tatu1247	Tatuyo	saphon
+2064	tuo	tuca1252	Tucano	saphon
+2065	tue	tuyu1244	Tuyuca	saphon
+2066	bao	waim1255	Waimaha	saphon
+2067	yui	yuru1263	Yurutí	saphon
+2068	guq	ache1246	Aché	saphon
+2069	aqz	akun1241	Akuntsú	saphon
+2070	adw	amun1246	Amundava	saphon
+2071	api	apia1248	Apiaká	saphon
+2072	awt	araw1273	Araweté	saphon
+2073	asn	xing1248	Asurini do Xingú	saphon
+2074	asu	toca1235	Asuriní do Tocantins	saphon
+2075	avv	avac1239	Avá-Canoeiro	saphon
+2076	awe	awet1244	Awetí	saphon
+2077	gui	east2555	Chiriguano	saphon
+2078	gui	east2555	Chiriguano	saphon
+2079	eme	emer1243	Emerillon	saphon
+2080	gvo	gavi1246	Gavião do Jiparaná	saphon
+2081	gvj	guaj1256	Guajá	saphon
+2082	gub	guaj1255	Guajajára	saphon
+2083	gun	mbya1239	Mbyá	saphon
+2084	gug	para1311	Paraguayan Guaraní	saphon
+2085	gyr	guar1292	Guarayu	saphon
+2086	jua	juma1249	Júma	saphon
+2087	jur	juru1256	Jurúna	saphon
+2088	urb	urub1250	Kaapor	saphon
+2089	kgk	kaiw1246	Kaiwá	saphon
+2090	kay	kama1373	Kamayurá	saphon
+2091	ktn	kari1311	Karitiâna	saphon
+2092	arr	karo1305	Karo	saphon
+2093	kyz	kaya1329	Kayabí	saphon
+2094	kyr	kuru1309	Kuruáya	saphon
+2095	mpu	maku1278	Makuráp	saphon
+2096	myu	mund1330	Mundurukú	saphon
+2097	yrl	nhen1239	Nheengatú	saphon
+2098	omg	omag1248	Omagua	saphon
+2099	pta	pait1247	Pai Tavytera	saphon
+2100	pak	para131
```

Comment:
good point. i've gone through all the codes now in mappings/InventoryID-LanguageCodes.tsv (i shortened the filename).

there are a couple of issues here.

1. before we had a few macrolanguage ISO codes, e.g.

2181	est	NA	Estonian	uz

i've updated those now so that we have the more precise ISO code [ekk] for Standard Estonian and added the Glottolog code

2. Sometimes Glottolog is more precise than ISO, for example here:

http://glottolog.org/resource/languoid/id/gunw1252

So I take the more precise Glottolog code and the parent node ISO code, e.g. Kune.

2657	gup	gund1246	Gun-Djeihmi	er
2658	gup	gune1238	Kune	er

3. Lastly, when the data is actually missing from Ethnologue, I leave it NA (but we can pick something else if you want).

2691	NA	mink1237	Minkin
2692	NA	ngum1253	Ngumbarl

In these cases it says explicitly in Glottolog that the entry is missing in ISO (happens often in extinct languages of Australia, for example).

4. When both ISO and Glottolog are NA it means there exists no code for that language yet. In  the cases of Aussie languages, they will probably get updated in Glottolog at some point and then I'll add those codes. I can't say anything regarding ISO and its update methodologies.

In a few other cases, e.g.

2264	NA	NA	Tod Tibetan	ea

I looked at some point at these, but I will revisit them in more detail. I remember there were a  few Chinese dialects.

Finally, I like our idea of keeping the codes in the original sources and dropping warnings when they don't match. I will also at some point revisit these NA NA languages.